### DISCUSSION TITLE
Loading design generated by innovus in openroad

### DISCUSSION THREAD

**Discussion by GundraHarshini** [The-OpenROAD-Project/OpenROAD#6416] [General]
I have read the lib,lef,tlef files and then def file, and then sdc,sdf files to do the timing analysis outside the innovus. Is it the right way to do it ? Or Iam missing anywhere because if a see a path using report_checks in openroad its timing,slack values are different from innovus generated timing reports.

**Comment by maliberty** [comment]
tlef is a proprietary format so not relevant.  Those files sound correct.  Do you get any errors loading the SDC?  You could also try in standalone sta as it supports hierarchical constraints better (for now).

**Reply by rovinski** [reply]
I think `.tlef` is a non-standard extension to designate tech LEF files. I'm pretty sure Sky130 uses it. I've also seen one proprietary PDK use it.

**Reply by maliberty** [reply]
Some people use tlef just to mean a standard technology LEF.  Sometimes I see people referring to .tf as tlef.

**Comment by GundraHarshini** [comment]
There were no errors in sdc but while reading sdf it shows like some hierarchical pins were not found. If I use link_design command before reading sdf then there were no such errors, but the design in the gui is missing as soon as I link the design. Can we open gui in standalone sta?

**Reply by maliberty** [reply]
You should link the design before reading sdf or sdc.  For timing you don't need the DEF but if you load it then it should come after linking.  Standalone sta has no gui.
